### PR TITLE
feat(authn): Adds SAML configuration support

### PR DESCRIPTION
**Pull request by ttomsu**

### CODE REVIEW

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/saml/EditSamlCommand.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+@Parameters(separators = "=")
+public class EditSamlCommand extends AbstractEditAuthnMethodCommand<Saml> {
+
+  @Getter
+  private AuthnMethod.Method method = AuthnMethod.Method.SAML;
+
+  @Parameter(
+      names = "--metadata",
+      description = "The address to your identity provider's metadata XML file. This can be a URL" +
```
Comment (lwander) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/saml/EditSamlCommand.java]: This needs to be prefixed with `http` or `file`, right? If so, can you call that out here?

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/saml/EditSamlCommand.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import com.netflix.spinnaker.halyard.config.model.v1.security.AuthnMethod;
+import com.netflix.spinnaker.halyard.config.model.v1.security.Saml;
+import lombok.Getter;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+@Parameters(separators = "=")
+public class EditSamlCommand extends AbstractEditAuthnMethodCommand<Saml> {
+
```
Comment (lwander) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/saml/EditSamlCommand.java]: Can you add a `String longDescription = ...` and `String shortDescription = ...` that describes saml in a little more detail?

https://github.com/spinnaker/halyard/issues/305

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/saml/EditSamlCommand.java (hunk 1 above)
Comment (ttomsu) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/saml/EditSamlCommand.java]: `http` is needed for a URL, but for a local file, only a path is needed, and the `file:` is prepended later.

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/saml/EditSamlCommand.java (hunk 2 above)
Comment (ttomsu) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/saml/EditSamlCommand.java]: Added.

**Pull request merged**